src/effective-dart/usage.md

```diff
@@ -350,13 +350,18 @@ then it probably does make sense to have a separate boolean field.
 
 Checking that a nullable variable is not equal to `null` promotes the variable
 to a non-nullable type. That lets you access members on the variable and pass it
-to functions expecting a non-nullable type. Unfortunately, promotion is only
-sound for local variables and parameters, so fields and top-level variables
-aren't promoted.
+to functions expecting a non-nullable type.
 
-One pattern to work around this is to assign the field's value to a local
-variable. Null checks on that variable do promote, so you can safely treat
-it as non-nullable.
+Type promotion is only sound, however, for local variables, parameters, and
```

Comment:
This is a little bit of a fib because type promotion _would_ be sound for top-level variables, provided that they're final. We just didn't implement type promotion for top-level variables because we didn't think it would be useful enough to justify the engineering effort.

Maybe we should just change "sound" to "supported"?